api_shadow/BUILD

```diff
@@ -0,0 +1,157 @@
+# DO NOT EDIT. This file is generated by tools/proto_sync.py.
```

Comment:
Should we maybe put this somewhere else that is not top level so that people don't see it, ignore the README, and get pretty confused? My first thought was a top level `generated/` directory, but I think we have scripts that put stuff in a directory by that name so that might get confusing. Also, it's still top level. Maybe somewhere under `bazel/` as it is a baked in part of our build that should eventually just go away? WDYT?

Reply:
How about `prebuillt/api_shadow`, `internal/api_shadow`, `vendored/api_shadow`? Prefer not under `bazel/`, as this doesn't technically relate to Bazel infra (but it is build related).

Reply:
Also, maybe `generated_api`?

Reply:
I kind of like `prebuilt/api_shadow` but I don't feel that strongly about it. (2nd place would be `vendored/`)

Reply:
I put it in `generated_api_shadow`; it's only 3 lines to change it to whatever folks want, so if @lizan has any strong preferences, we can do that.